sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/TestConfigurableSpanExporterProvider.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurableSpanExporterProvider;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.Collection;
+
+public class TestConfigurableSpanExporterProvider implements ConfigurableSpanExporterProvider {
+  @Override
+  public SpanExporter createExporter(ConfigProperties config) {
+    return new TestSpanExporter(config.getBoolean("should.always.fail"));
+  }
+
+  @Override
+  public String getName() {
+    return "testExporter";
+  }
+
+  public static class TestSpanExporter implements SpanExporter {
+
+    private final boolean shouldAlwaysFail;
+
+    public TestSpanExporter(boolean shouldAlwaysFail) {
+      this.shouldAlwaysFail = shouldAlwaysFail;
+    }
+
+    @Override
+    public CompletableResultCode export(Collection<SpanData> spans) {
+      return shouldAlwaysFail
+          ? CompletableResultCode.ofFailure()
+          : CompletableResultCode.ofSuccess();
+    }
+
+    @Override
+    public CompletableResultCode flush() {
+      return shouldAlwaysFail
+          ? CompletableResultCode.ofFailure()
+          : CompletableResultCode.ofSuccess();
+    }
+
+    @Override
+    public CompletableResultCode shutdown() {
+      return shouldAlwaysFail
+          ? CompletableResultCode.ofFailure()
+          : CompletableResultCode.ofSuccess();
```

Comment:
factor out private internal method to eliminate this duplication?

Reply:
sure...I was pondering ripping this out and just exposing the ConfigProperties on this test class which seems a little less obtuse and simpler.

Reply:
ripped this stuff out and made the test simpler. please take a look.

Reply:
Seems more straightforward now.  Is was testing the config through the methods, now it can just get the config.  Simpler.